src/Microsoft.AspNetCore.OData/Builder/ODataConventionModelBuilder.cs

```diff
@@ -55,14 +64,44 @@ public ODataConventionModelBuilder(IServiceProvider provider)
         /// the functionality is due to the way assembly resolution is done in AspNet vs AspnetCore.
         /// </remarks>
         public ODataConventionModelBuilder(IServiceProvider provider, bool isQueryCompositionMode)
+            : this(provider, null, isQueryCompositionMode)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ODataConventionModelBuilder"/> class.
+        /// </summary>
+        /// <param name="provider">The service provider to use.</param>
+        /// <param name="applicationPartManager">
+        /// The application part manager to use. If null, the service
+        /// provider will be queried for the application part manager.
+        /// </param>
+        /// <param name="isQueryCompositionMode">If the model is being built for only querying.</param>
+        /// <remarks>
+        /// The model built if <paramref name="isQueryCompositionMode"/> is <c>true</c> has more relaxed
+        /// inference rules and also treats all types as entity types. This constructor is intended for use
+        /// by unit testing only.
```

Comment:
The fact that this constructor is only for unit testing should potentially be in the summary after the current text.